modin/core/dataframe/pandas/metadata/index.py

```diff
@@ -70,6 +141,63 @@ def get(self, return_lengths=False) -> pandas.Index:
         else:
             return self._value
 
+    def equals(self, other: "ModinIndex") -> bool:
```

Comment:
is there a reason _not_ to implement a magic `__eq__` instead of custom-named `equals()`?.. Python is not Java, you know...

Reply:
the `__eq__` operator doesn't mean the same as `.equals()` in the context of pandas indices, so first of all I was willing to be aligned with this logic:
```python
>>> pandas.Index([1, 2]) == pandas.Index([1, 2])
array([ True,  True])
>>> pandas.Index([1, 2]).equals(pandas.Index([1, 2]))
True
```

Then there is [this logic in `ModinIndex.__getattr__`](https://github.com/modin-project/modin/blob/5dee270d5b8e7896172e941e8faba2737f4e2016/modin/core/dataframe/pandas/metadata/index.py#L107-L131), which makes me think that `ModinIndex` is some kind of an extension for the regular `pandas.Index`, meaning that all the methods that intersect with the `pandas.Index` API should have the same behavior (so I have to call the method `.equals()` and not `.__eq__()` to perform the comparison in the way I wanted)